packages/site/src/lib/components/Slides.svelte

```diff
@@ -1,35 +1,66 @@
 <script lang="ts">
   export let events: Event[];
+
   import Image from "$lib/components/Image.svelte";
+  import ChevronRight from "@material-symbols/svg-400/rounded/chevron_right.svg?component";
+  import ChevronLeft from "@material-symbols/svg-400/rounded/chevron_left.svg?component";
+
+  interface Poster {
+    altText: string;
+    imageUrl: string;
+  }
 
   interface Event {
-    poster?: {
-      altText: string;
-      imageUrl: string;
-    };
+    poster?: Poster
+  }
+
+  const posters = (): Poster[] => events.flatMap((ev) => ev.poster ? [ev.poster] : [])
+
+  let currentIdx = 0;
+  let current: Poster | undefined = posters()[0]
```

Comment:
```suggestion
  let currentIdx = 0;
  $: current: Poster | undefined = currentIdx >= 0 && currentIdx < posters.length ? posters[currentIdx] : undefined
```

[Svelte keeps derived data like this up-to-date with its reactive declaration syntax](https://svelte.dev/tutorial/reactive-declarations). Anytime `currentIdx` is updated, `current` will be updated automatically.

Also moved the bounds check into here just to make sure `currentIdx` is in bounds (if the events array has zero elements then `current` should be `undefined`).

Reply:
Ok, thanks for explaining that to me, had no idea what the $: syntax meant at first